Sources/NIOSSL/NIOSSLClientHandler.swift

```diff
@@ -14,17 +14,30 @@
 
 import NIO
 
-private extension String {
-    func isIPAddress() -> Bool {
+internal func validateSNIServerName(_ hostname: String) throws {
```

Comment:
Can this become a function as an extension on `String`? I kinda hate unnecessary top-level functions.

Reply:
sure. done